internal/app/handlers/saveurljson/saveurljson.go

```diff
@@ -68,8 +72,11 @@ func PostURLjson(log *slog.Logger, storage URLSaverJSON) http.HandlerFunc {
 			return
 		}
 
-		alias = saveurl.NewRandomString(8) //поправить
-		if _, ok := storage.GetStorageURL(alias); ok {
+		alias = utilitys.NewRandomString(config.AliasLength) //поправить
+
+		ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
```

Comment:
```suggestion
		ctx, cancel := context.WithTimeout(r.Context()), 2*time.Second)
```
Контекст для всех процедур, которые необходимо уместить в работу хендлера, необходимо брать от запроса. На него при необходимости можно сверху повесить еще и таймаут.